src/cirrus/lib2/events.py

```diff
@@ -0,0 +1,249 @@
+        statedb: StateDB = None,
+        eventdb: EventDB = None,
+        batch_size: int = 10,
+    ):
+        self.logger = logger
+        self._boto3_session = boto3_session if boto3_session else boto3.Session()
+        wf_event_topic_arn = os.getenv("CIRRUS_WORKFLOW_EVENT_TOPIC_ARN")
+        self.event_publisher = (
+            SNSPublisher(
+                wf_event_topic_arn,
+                logger=logger,
+                boto3_session=self._boto3_session,
+                batch_size=batch_size,
+            )
+            if wf_event_topic_arn
+            else None
+        )
+        self.statedb = (
+            statedb if statedb is not None else StateDB(session=self._boto3_session)
+        )
+        self.eventdb = (
+            eventdb if eventdb is not None else EventDB(session=self._boto3_session)
+        )
+
+    def flush(self):
+        if self.event_publisher:
+            self.event_publisher.execute()
+
+    @classmethod
+    @contextmanager
+    def handler(
+        cls: "WorkflowEventManager", *args, **kwargs
+    ) -> AbstractContextManager["WorkflowEventManager"]:
+        wfem = cls(*args, **kwargs)
+        try:
+            yield wfem
+        finally:
+            wfem.flush()
+
+    def announce(
+        self,
+        event_type: WFEventType,
+        payload: dict = None,
+        payload_id: str = None,
+        extra_message: dict = None,
+    ) -> None:
+        """
+        Construct message payload and publish to WorkflowEventTopic.
+
+        Args:
+            event_type (WFEventType): the type of event which occurred
+            payload (dict): The ProcessPayload which triggered the event.
+            payload_id (str): The ID of the payload.  Must match payload["id"] if that
+                              exists / is  provided.
+            extra_message (dict): dictionary of additional items to be placed into the
+                                  SNS message body.
+        """
+        if self.event_publisher is None:
+            return
+
+        if payload is None:
+            if payload_id is None:
+                raise ValueError("must specify payload_id or payload")
+        elif payload_id is None:
+            payload_id = payload["id"]
+
+        if payload is not None and payload["id"] != payload_id:
+            raise ValueError(
+                "payload_id and payload['id'] must match if both supplied."
+            )
+        message = {
+            "event_type": event_type,
+            "payload": payload,
+            "payload_id": payload_id,
+        }
+
+        if extra_message:
+            if not all(k not in message for k in extra_message):
+                raise ValueError(
+                    "extra_message parameters must not include: "
+                    + f"{','.join(message.keys())}."
+                )
+            message.update(extra_message)
+        self.event_publisher.add(
+            message=json.dumps(message),
+            message_attrs={
```

Comment:
I think this was mentioned before, but adding things like the payload ID, collections, etc, could be of benefit to downstream subscribers. We should discuss what all we can add here.